subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/execution/PartialEvaluator.kt

```diff
@@ -30,108 +33,143 @@ enum class ProgramKind {
 
 enum class ProgramTarget {
     Project,
-    Settings
+    Settings,
+    Gradle
```

Comment:
What is `Gradle` in this? Also, what about `InitScript` support?

Reply:
From the point of the view of the Kotlin DSL, an init script is script that targets the `Gradle` singleton.